fix for SpecifySPMModel

MartinPerez opened the pull request:
@satra you were right, I tested simply correcting the line you pointed out solved the problem. Fix for #1198.

satra commented on nipype/algorithms/modelgen.py:
```diff
@@ -476,7 +476,7 @@ def _concatenate_info(self, infolist):
                         infoout.onsets[j].extend(onsets.tolist())
                 for j, val in enumerate(info.durations):
-                    if len(val) > 1:
+                    if len(val) > 0:
```
i think this can simply be: `if len(val):`

also could you please add your test to test_modelgen? it should fail without this change, and pass with it? there is already a section that checks concatenation.

satra commented on nipype/algorithms/modelgen.py:
```diff
@@ -497,7 +497,7 @@ def _concatenate_info(self, infolist):
             onelist[0, sum(nscans[0:i]):sum(nscans[0:(i + 1)])] = 1
             infoout.regressors.insert(len(infoout.regressors),
-                                      onelist.tolist()[0])
+                                      onelist.tolist()[0])        
```
extra white space added.

MartinPerez commented:
@satra I implemented the test and corrected based on your comments. In the test I added some comments to separate the intention of the different tests and added mine at the end. Also modified the last test to contrast it with mine when there are three sessions and they are not concatenated the durations are detected just fine for all runs.

satra commented:
thanks @MartinPerez

MartinPerez commented:
thanks @satra

chrisfilo commented:
CircleCI tests are not passing on this PR. We should consider reverting the merge until this is resolved.

satra commented:
@chrisfilo - is there an error message? this should only affect the spm test.

MartinPerez commented:
Effectively this should be reverted. The key problem is the same that I had (non matching number of onsets and events / Number of event onsets (16) does not match the number of durations (4)) but i guess now its about specifying all durations with only one number?

satra commented:
i think we should fix it to address both cases. 

```
if len(val) == 1:
    # then replicate to number of onsets
else:
    if len(val) != len(number_onsets):
        raise ValueError('Mismatch in number of onsets and durations for session %d, condition %s')
    # append all durations
```

MartinPerez commented:
I would say that we have to check the number of onsets, like:

if len(ons) > 1 and len(val) == 1:
# fill ons with val
elif len(ons) ==  len(val):
# add the durations
else:
    raise ValueError('Mismatch in number of onsets and durations for session %d, condition %s')

you agree?
